mlflow/models/utils.py

```diff
@@ -740,6 +808,10 @@ def _is_scalar(x):
             actual_cols = expected_required_cols
         elif isinstance(pf_input, pd.DataFrame):
             actual_cols = set(pf_input.columns)
+            # for schemas with a single column, match input with column
+            if len(actual_cols) == 1:
+                pf_input.rename(columns={actual_cols.pop(): input_names[0]}, inplace=True)
+                actual_cols = expected_required_cols
```

Comment:
why do we need this change?

Reply:
Because if we infer signature from input_example like '{"model_input": ["input"]}', it would be Schema([ColSpec(DataType.string, "model_input")]). While when we pass ["input"] to the model for predict `pyfunc_model.predict(["input"])` it will convert the list into a pandas dataframe, and check if the column name matches the schema's name. So I think for this kind of situation we should just match the input name directly, like what we do for np.ndarray above.

Reply:
what happens if we pass `pyfunc_model.predict(pd.DataFrame({"foo": ["a", "b", "c"]}))`?

Reply:
This would be a dataframe with column 'foo' and three rows 'a', 'b', 'c' separately. It accepts the column.
I think this behavior is because we always turn input into pandas dataframe, which sometimes doesn't make sense for non-pandas input.

Reply:
does `pd.DataFrame({"foo": ["a", "b", "c"]})` pass the schema validation?